pgjdbc/src/main/java/org/postgresql/hostchooser/HostRequirement.java

```diff
@@ -31,4 +31,16 @@ public boolean allowConnectingTo(HostStatus status) {
   };
 
   public abstract boolean allowConnectingTo(HostStatus status);
+
+  /**
+   *
+   * @param targetServerType
+   * @return
+   */
+
+  public static HostRequirement getTargetServerType(String targetServerType) {
+    String allowSlave = targetServerType.replaceFirst("slave", targetServerType);
```

Comment:
I suggest `targetServerType.replace('lave', 'econdary')` to support both `slave` and `preferSlave` in one go.